homeassistant/components/watchyourlan/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Config flow for WatchYourLAN integration."""
+
+import logging
+from typing import Any
+
+from httpx import ConnectError, HTTPStatusError
+import voluptuous as vol
+from watchyourlanclient import WatchYourLANClient
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_URL, CONF_VERIFY_SSL
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# Schema for the user setup form
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): str,
+        vol.Optional(CONF_VERIFY_SSL, default=False): bool,
+    }
+)
+
+
+class WatchYourLANConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WatchYourLAN."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            # Set unique ID (use the URL as the unique ID)
+            await self.async_set_unique_id(user_input[CONF_URL])
+            # Abort if the unique ID is already configured
+            self._abort_if_unique_id_configured()
```

Comment:
the url is a bad unique_id as it can change, please use `self._abort_entries_match` instead and don't set a unique id